packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1066,14 +1072,45 @@ class CupertinoSliverNavigationBar extends StatefulWidget {
 // A state class exists for the nav bar so that the keys of its sub-components
 // don't change when rebuilding the nav bar, causing the sub-components to
 // lose their own states.
-class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigationBar> {
+class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigationBar> with TickerProviderStateMixin {
   late _NavigationBarStaticComponentsKeys keys;
   ScrollableState? _scrollableState;
+  Widget? effectiveBottom;
+  Widget? effectiveLeading;
+  Widget? effectiveTrailing;
+  bool? effectiveStretch;
+  NavigationBarBottomMode? effectiveBottomMode;
+  late AnimationController _animationController;
+  late AnimationController _fadeController;
+  Tween<double> persistentHeightTween = Tween<double>(
+    begin: _kNavBarPersistentHeight,
+    end: 0.0,
+  );
+  Tween<double> largeTitleHeightTween = Tween<double>(
+    begin: _kNavBarLargeTitleHeightExtension,
+    end: 0.0,
+  );
+  late Animation<double> persistentHeightAnimation;
+  late Animation<double> largeTitleHeightAnimation;
+  bool atTop = false;
 
   @override
   void initState() {
     super.initState();
     keys = _NavigationBarStaticComponentsKeys();
+    _animationController = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 300),
+    );
+    _fadeController = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 150),
+    );
+    persistentHeightAnimation = persistentHeightTween.animate(_animationController)
+      ..addListener(() {
+        setState(() { });
```

Comment:
The issue starts with the `middle` widget. WHen the search bar animates to the top, the `middle` widget is briefly shown:

https://github.com/user-attachments/assets/b9281e95-0527-43e0-a502-9c9af0e2c027

To prevent that, I tried replacing the `middle` with an empty string so nothing shows up while the search bar is animating:

```dart
userMiddle: _animationController.isAnimating ? const Text('') : widget.middle,
```

But what then happens without the setState listener is that the middle widget is shown initially, but once the search bar animation happens the empty string is now always shown:


https://github.com/user-attachments/assets/74755f40-1645-4cef-b4ac-7b8467b3dc47